anomalib/data/mvtec.py

```diff
@@ -415,47 +424,45 @@ def __init__(
         self.test_batch_size = test_batch_size
         self.num_workers = num_workers
 
-        self.train_data: Dataset
-        self.val_data: Dataset
+        self.create_validation_set = create_validation_set
+        self.seed = seed
 
-    def prepare_data(self):
-        """Prepare MVTec Dataset."""
-        # Train
-        MVTec(
-            root=self.root,
-            category=self.category,
-            pre_process=self.pre_process,
-            is_train=True,
-            download=True,
-        )
-
-        # Test
-        MVTec(
-            root=self.root,
-            category=self.category,
-            pre_process=self.pre_process,
-            is_train=False,
-            download=True,
-        )
+        self.train_data: Dataset
+        self.test_data: Dataset
+        if create_validation_set:
+            self.val_data: Dataset
```

Comment:
Is flake and pylint ok with this? When we don't define the variables in `__init__`, they usually complain.